stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetActivity.kt

```diff
@@ -272,19 +296,32 @@ internal class PaymentSheetActivity : BasePaymentSheetActivity<PaymentResult>()
         viewModel.updateMode(transitionTarget.sheetMode)
     }
 
-    private fun setupBuyButton() {
-        viewBinding.buyButton.completedAnimation.observe(this) { completedState ->
-            completedState?.paymentIntentResult?.let { paymentIntentResult ->
-                onActionCompleted(paymentIntentResult)
-            }
-        }
+    private val viewStateObserver: (ViewState.PaymentSheet?) -> Unit
+        get() = { viewState ->
+            lifecycleScope.launch {
```

Comment:
Hmmm, is this going to be a case of doing too much work on the UI thread?  Maybe a different scope is needed?

Reply:
All animation work needs to happen on the UI thread.